File: src/main/java/au/com/anthonybruno/gymbuddy/workout/Workout.java
package au.com.anthonybruno.gymbuddy.workout;

import java.time.Duration;
import java.util.Date;

public class Workout {

    private final int id;
    private final Date date;
    private final Duration duration;
    private final String title;
    private final String description;

    public Workout(int id, Date date, Duration duration, String title, String description) {
        this.id = id;
        this.date = date;
        this.duration = duration;
        this.title = title;
        this.description = description;
    }

    public int getId() {
        return id;
    }

    public Date getDate() {
        return date;
    }

    public Duration getDuration() {
        return duration;
    }

    public String getTitle() {
        return title;
    }

    public String getDescription() {
        return description;
    }
}

File: backend/src/main/kotlin/dev/anthonybruno/gymbuddy/workout/Workout.kt
package dev.anthonybruno.gymbuddy.workout

import java.time.Instant

data class Workout(val id: Int,
                   val date: Instant,
                   val title: String?,
                   val description: String?,
                   val exercises: List<WorkoutExerciseDto>)

data class WorkoutExerciseDto(val name: String, val weight: Int?, val sets: Int, val reps: Int)

data class Exercise(val id: Int, val name: String, val description: String?, val mainMuscle: String)

data class AddWorkout(val title: String?,
                      val description: String?,
                      val exercises: List<AddExercise>)

data class AddExercise(val id: Int, val sets: Int, val reps: Int, val weight: Int?)

data class WorkoutStats(
        val lastWorkout: Instant?,
        val commonExercise: String,
        val workoutsLast30Days: Int
)
